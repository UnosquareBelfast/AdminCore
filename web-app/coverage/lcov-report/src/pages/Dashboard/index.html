<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/pages/Dashboard</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="..\..\..\prettify.css" />
    <link rel="stylesheet" href="..\..\..\base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(..\..\..\sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="..\..\..\index.html">All files</a> src/pages/Dashboard
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">11.11% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>7/63</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/15</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">3.85% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>1/26</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">11.11% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>7/63</span>
      </div>
    </div>
    <p class="quiet">
      Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
    </p>
  </div>
  <div class='status-line low'></div>
<div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file low" data-value="container.js"><a href="container.js.html">container.js</a></td>
	<td data-value="3.51" class="pic low"><div class="chart"><div class="cover-fill" style="width: 3%;"></div><div class="cover-empty" style="width:97%;"></div></div></td>
	<td data-value="3.51" class="pct low">3.51%</td>
	<td data-value="57" class="abs low">2/57</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="15" class="abs low">0/15</td>
	<td data-value="4" class="pct low">4%</td>
	<td data-value="25" class="abs low">1/25</td>
	<td data-value="3.51" class="pct low">3.51%</td>
	<td data-value="57" class="abs low">2/57</td>
	</tr>

<tr>
	<td class="file high" data-value="index.jsx"><a href="index.jsx.html">index.jsx</a></td>
	<td data-value="83.33" class="pic high"><div class="chart"><div class="cover-fill" style="width: 83%;"></div><div class="cover-empty" style="width:17%;"></div></div></td>
	<td data-value="83.33" class="pct high">83.33%</td>
	<td data-value="6" class="abs high">5/6</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="1" class="abs low">0/1</td>
	<td data-value="83.33" class="pct high">83.33%</td>
	<td data-value="6" class="abs high">5/6</td>
	</tr>

</tbody>
</table>
</div><div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Tue Jun 19 2018 15:12:10 GMT+0100 (GMT Summer Time)
</div>
</div>
<script src="..\..\..\prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="..\..\..\sorter.js"></script>
<script src="..\..\..\block-navigation.js"></script>
</body>
</html>
