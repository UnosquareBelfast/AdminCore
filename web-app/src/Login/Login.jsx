import React from 'react';
import Swal from 'sweetalert2';
import styles from './Login.css';
import LoginService from './LoginService';

class Login extends React.Component  {
    constructor(){
        super();
        this.handleChange = this.handleChange.bind(this);
        this.handleSubmit = this.handleSubmit.bind(this);

        this.Auth = new LoginService();
    }
    render() {
        return (
            <div className={styles.center}>
                <div className={styles.card}>
                    <h1>Login</h1>
                    <form onSubmit={this.handleSubmit}>
                        <input className={styles.item} placeholder="Email goes here..." name="email" type="text" onChange={this.handleChange} />
                        <input className={styles.item} placeholder="Password goes here..." name="password" type="password" onChange={this.handleChange} />
                        <input className={styles.submit} value="SUBMIT" type="submit" />
                    </form>
                </div>
            </div>
        );
    }

    componentWillMount(){
        if(this.Auth.loggedIn())
            this.props.history.replace('/');
    }

    handleChange(e){
        this.setState(
            {
                [e.target.name]: e.target.value
            }
        )
    }

    handleSubmit(e){
        e.preventDefault();
      
        this.Auth.login(this.state.email,this.state.password)
            .then(response =>{
               this.props.history.replace('/');
            })
            .catch(error =>{
                Swal('Could not log in', error.message, 'error');
            })
    }
}

export default Login;