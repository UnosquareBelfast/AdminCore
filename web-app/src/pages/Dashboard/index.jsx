import React from 'react';
import { PropTypes as PT } from 'prop-types';
import Swal from 'sweetalert2';
import styles from './style.css';
import Calendar from 'react-calendar';
import Moment from 'moment';
import DashboardService from './DashboardService.js';
import {
  Layout,
  TakenLeave,
  BookedLeave,
  RequestHoliday,
} from '../../components';
import withAuth from '../../hoc/withAuth';

class Dashboard extends React.Component {
  static propTypes = {
    user: PT.object,
  }

  constructor(props) {
    super(props);

    this.state = {
      date: Moment(),
      daysRemaining: 0,
      requestModalOpen: false,
      user: null,
    };

    this.toggleHolidayModal = this.toggleHolidayModal.bind(this);
    this.DashboardService = new DashboardService();
  }

  componentDidMount() {
    //this.getUserProfile();
    // this.DashboardService.getHolidays(this.props.user.id)
    //         .then(response =>{
    //           const remains = 23 - response.length;
    //           this.setState({ daysRemaining: remains});
    //         })
    //         .catch(error =>{
    //             Swal('Could not log in', error.message, 'error');
    //         })
    this.getUserProfile();
  }

  getUserProfile() {
    const email = this.DashboardService.getUserEmail();

    const splitOne = email.split('.');
    const splitTwo = splitOne[1].split('@');
    const firstName = splitOne[0];
    const lastName = splitTwo[0];

    this.DashboardService.getUserProfile(firstName, lastName)
      .then(response =>{
        // eslint-disable-next-line
        console.log(response);
        this.setState({user : response});
      })
      .catch(error =>{
        Swal({title: 'Could not get user profile', text: error.message, type: 'error'});
      });
  }

  getHolidays() {

  }

  toggleHolidayModal() {
    this.setState({
      requestModalOpen: !this.state.requestModalOpen,
    });
  }


  render() {
    return (
      <Layout {...this.props}>
        <div className={styles.RowC}>
          <div className={styles.LeaveTrackerList}>
            <span>Days Taken:</span>
            <TakenLeave user={this.props.user} />
            <br/>
            <span>Days Booked:</span>
            <BookedLeave user={this.props.user} />
          </div>
          <div className={styles.CalendarDiv}>
            <span>Days Remaining: {this.state.daysRemaining}</span>

            <Calendar className={styles.CalendarStyling}
              onChangeMonth={(date) => this.setState({ date })}
              date={this.state.date}
              // eslint-disable-next-line
            onPickDate={(date => console.log(date))}
            />
            <br/>
            <button onClick={this.toggleHolidayModal}>Book Holiday</button>
            <RequestHoliday
              user={this.props.user}
              show={this.state.requestModalOpen}
              onClose={this.toggleHolidayModal} />
          </div>
        </div>
      </Layout>
    );
  }
}

export default withAuth(Dashboard);
