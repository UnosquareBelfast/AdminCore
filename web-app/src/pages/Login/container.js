import React from 'react';
import Swal from 'sweetalert2';
import { PropTypes as PT } from 'prop-types';
import { userLogin } from '../../services/userService';
import { isLoggedIn } from '../../utilities/currentUser';

export default Wrapped =>
  class extends React.Component {
    static propTypes = {
      history: PT.object,
    };
    constructor(props) {
      super(props);
      this.state = {
        email: '',
        password: '',
      };
      // this.state = {
      //   form: this.initialFormState(),
      //   formIsValid: false,
      //   loading: false,
      // };
    }

    componentWillMount() {
      if (isLoggedIn()) this.props.history.replace('/');
    }

    // initialFormState() {
    //   return {
    //     email: {
    //       label: 'Email:',
    //       elementType: 'input',
    //       elementConfig: {
    //         type: 'email',
    //         placeholder: 'Enter your email address',
    //       },
    //       value: '',
    //       validation: {
    //         required: true,
    //         isEmail: true,
    //       },
    //       valid: false,
    //       touched: false,
    //     },
    //     password: {
    //       label: 'Password:',
    //       elementType: 'input',
    //       elementConfig: {
    //         type: 'password',
    //         placeholder: 'Enter your password',
    //       },
    //       value: '',
    //       validation: {
    //         required: true,
    //         minLength: 4,
    //       },
    //       valid: false,
    //       touched: false,
    //     },
    //   };
    // }

    // handleFormChange = (event, inputIdentifier) => {
    //   const inputValue = event.target.value;
    //   const updatedForm = updateFormDataOnChange(
    //     this.state.form,
    //     inputIdentifier,
    //     inputValue,
    //   );

    //   let formIsValid = isFormValidOnChange(updatedForm);
    //   this.setState({ form: updatedForm, formIsValid: formIsValid });
    // };

    handleFormSubmit = e => {
      e.preventDefault();
      console.log('e :', e);
      //const formData = getFormDataOnSubmit(this.state.form);

      // userLogin(formData.email, formData.password)
      //   .then(() => {
      //     this.props.history.push('/');
      //   })
      //   .catch(error => {
      //     Swal({
      //       title: 'Could not log in',
      //       text: error.message,
      //       type: 'error',
      //     });
      //   });
    };

    render() {
      return (
        <Wrapped
          formIsValid={this.state.formIsValid}
          submitForm={this.handleFormSubmit}
          formChanged={(event, id) => this.handleFormChange(event, id)}
          loading={this.state.loading}
        />
      );
    }
  };
