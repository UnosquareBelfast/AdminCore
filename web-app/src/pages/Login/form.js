import React, { Component } from 'react';
import { PropTypes as PT } from 'prop-types';

export class Form extends Component {
  static propTypes = {
    submitForm: PT.func.isRequired,
    children: PT.array.isRequired,
  };

  constructor(props) {
    super(props);
    this.state = {
      invalidInputs: [],
      formIsValid: true,
    };
  }

  handleCheckInputValid = (label, isValid) => {
    let invalidInputs = [...this.state.invalidInputs];
    if (isValid) {
      if (invalidInputs.includes(label)) {
        let index = invalidInputs.indexOf(label);
        invalidInputs = invalidInputs.splice(index, 1);
      }
    } else {
      if (!invalidInputs.includes(label)) {
        invalidInputs.push(label);
      }
    }
    let formIsValid = invalidInputs.length > 0 ? false : true;
    this.setState({ formIsValid, invalidInputs });
  };

  render() {
    const childWithProp = React.Children.map(this.props.children, child => {
      return React.cloneElement(child, {
        validateForm: this.handleCheckInputValid,
      });
    });

    return (
      <form id="loginForm" onSubmit={this.props.submitForm}>
        {childWithProp} {/* instead of this.props.children */}
        <button type="submit" disabled={this.formIsValid}>
          Login
        </button>
      </form>
    );
  }
}
export default Form;
