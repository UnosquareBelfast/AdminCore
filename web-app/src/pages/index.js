export { default as Dashboard } from './Dashboard';
export { default as Login } from './Login';
export { default as Admin } from './Admin';
export { default as User } from './User';
export { default as Profile } from './Profile';
