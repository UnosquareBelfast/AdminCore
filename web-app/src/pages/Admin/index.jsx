import React from 'react';
import { PropTypes as PT } from 'prop-types';
import { Route, Switch } from 'react-router-dom';
import container from './container';
import {
  CreateUser,
  UserListing,
  PendingHolidays,
  AllHolidays,
} from '../../components';
import { Card } from '../../components/common';
import Sidebar from './Sidebar';
import { Layout, withAuth } from '../../hoc';
import { flowRight } from 'lodash';
import {
  SidebarContainer,
  Container,
  MainContentContainer,
  Refresh,
} from './styled';

export const Admin = props => (
  <Layout {...props}>
    <Container>
      <SidebarContainer>
        <Sidebar />
      </SidebarContainer>
      <MainContentContainer>
        <Switch>
          <Route path="/admin/createEmployee" component={CreateUser} />
          <Route
            path="/admin/employees"
            render={() => (
              <Card>
                <h3>Employees</h3>
                <Refresh onClick={props.refreshUsers}>Refresh</Refresh>
                <UserListing users={props.users} />
              </Card>
            )}
          />
          <Route path="/admin/pendingHolidays" component={PendingHolidays} />
          <Route path="/admin/holidays" component={AllHolidays} />
        </Switch>
      </MainContentContainer>
    </Container>
  </Layout>
);

Admin.propTypes = {
  userDetails: PT.object,
  users: PT.array,
  refreshUsers: PT.func,
  history: PT.object,
};

const enhance = flowRight(
  withAuth,
  container,
);
export default enhance(Admin);
