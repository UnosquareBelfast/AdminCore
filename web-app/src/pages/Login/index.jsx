import React from 'react';
import { PropTypes as PT } from 'prop-types';
import { LoginBG, LoginPanel } from './styled';
import container from './container';
import { Input, Button } from '../../components/common';

export const Login = props => {
  const { formElementsArray, formIsValid, submitForm, formChanged } = props;

  let form = (
    <form id="loginForm">
      {formElementsArray.map(({ id, config }, index) => (
        <Input
          key={id}
          label={config.label}
          elementType={config.elementType}
          elementConfig={config.elementConfig}
          value={config.value}
          invalid={!config.valid}
          shouldValidate={config.validation}
          focus={index === 0 ? true : false}
          touched={config.touched}
          changed={event => formChanged(event, id)}
        />
      ))}
      <Button
        id="createUserBtn"
        label="Login"
        onClick={submitForm}
        disabled={!formIsValid}
      />
    </form>
  );

  return (
    <LoginBG>
      <LoginPanel>
        <h1>Welcome to Admin Core</h1>
        {form}
      </LoginPanel>
    </LoginBG>
  );
};

Login.propTypes = {
  formElementsArray: PT.array.isRequired,
  formIsValid: PT.bool.isRequired,
  submitForm: PT.func.isRequired,
  formChanged: PT.func.isRequired,
};

export default container(Login);
