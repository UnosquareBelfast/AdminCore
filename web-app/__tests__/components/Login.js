import React from 'react';
import { shallow } from 'enzyme';
import { Login } from '../../src/pages/Login';
const mockHandleEvent = jest.fn();

afterEach(() => {
  mockHandleEvent.mockReset();
});

describe('Login Component', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<Login handleChange={mockHandleEvent} />);
    expect(wrapper.exists()).toBe(true);
    expect(wrapper).toMatchSnapshot();
  });
  it('onChange of email input calls handleChange', () => {
    const wrapper = shallow(<Login handleChange={mockHandleEvent} />);
    let emailInput = wrapper.find('[name="email"]');
    
    expect(mockHandleEvent.mock.calls.length).toEqual(0);    
    emailInput.simulate('change');
    expect(mockHandleEvent.mock.calls.length).toEqual(1);
  });
  it('onChange of password input calls handleChange', () => {

    const wrapper = shallow(<Login handleChange={mockHandleEvent} />);
    let passwordInput = wrapper.find('[name="password"]');

    expect(mockHandleEvent.mock.calls.length).toEqual(0);    
    passwordInput.simulate('change');
    expect(mockHandleEvent.mock.calls.length).toEqual(1);
  });
  it('on Form Submit handleSubmit is called', () => {

    const wrapper = shallow(<Login handleSubmit={mockHandleEvent} />);
    let form = wrapper.find('form');

    expect(mockHandleEvent.mock.calls.length).toEqual(0);    
    form.simulate('submit');
    expect(mockHandleEvent.mock.calls.length).toEqual(1);
  });
});
