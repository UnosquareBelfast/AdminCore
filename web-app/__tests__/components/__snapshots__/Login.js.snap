// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Component renders correctly 1`] = `
<div
  className="center"
>
  <div
    className="card"
  >
    <h1>
      Login
    </h1>
    <form
      id="loginForm"
    >
      <input
        className="item"
        name="email"
        onChange={[MockFunction]}
        placeholder="Email goes here..."
        type="text"
      />
      <input
        className="item"
        name="password"
        onChange={[MockFunction]}
        placeholder="Password goes here..."
        type="password"
      />
      <input
        className="submit"
        type="submit"
        value="SUBMIT"
      />
    </form>
  </div>
</div>
`;
